mezo-org/mezod
# Implement `cleanupApplications` function for the ValidatorPool

Closes: https://linear.app/thesis-co/issue/ENG-311/add-the-validatorpoolcleanupapplications-function

### Introduction

Adds a cleanupApplications function to the ValidatorPool precompile. When called by the contract owner, this function removes all pending applications. 

### Changes

Adds `CleanupApplications` to the poa keeper

Adds `CleanupApplicationsMethod` to ValidatorPool

Adds `ApplicationsCleanedEvent` to ValidatorPool

Updates ValidatorPool's interface, caller contract and abi.json

Updates ValidatorPool's `EvmByteCode` value using newly compiled artifacts

Adds hardhat task `validatorPool:cleanupApplications`

### Testing

On a localnet using hardhat:

1) Submit valid application(s)
2) Use the `validatorPool:applications` task to list pending applications
3) Use the `validatorPool:cleanupApplications` task to remove all pending applications (with owner as signer)
4) Use the `validatorPool:applications` task to verify there are no applications

---

### Author's checklist


### Reviewer's checklist
